### PR TITLE
WIP - Feature/umbraco8 di refactor

### PR DESCRIPTION
**Pull request by euanrae**
Hi,

TLDR
I've been working on this on and off, and I've got the registration working OK with DI, but the structure for the mapping work feels wrong, if you have any time to give some advice on how to restructure it better, I'd really appreciate it

Longer version
I've been playing around with this for a while, and I think the refactoring for initialising the mappers is OK.  UmbMapperRegistry now has an interface and should be registered as a Singleton in the DI container.  I've added an initializer class which is now used for registration of mappers.

I've also create new class in the 'Models' directory to wrap existing funtionality to allow registration to work with DI and keep fluent mapping API.

I'm having trouble restructuring the mapping parts nicely.  Functionally, it works as expected, though it needs a lot more test coverage.  The part that doesn't seem right is the way the UmbMapperService has to pass instances of itself around to factories that create mapping processors that need the IUmbMapperService.

I appreciate you're busy and this isn't your biggest priority, but if you have any time to look at it, any tips on how to sort out the structure, etc, I'd really appreciate it.

Thanks

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
Thanks for working on this. I'll try and have a look this weekend.
